### ISSUE TITLE
"Code" formatting ("pre" tag) is stripping new lines

### ISSUE THREAD

**Issue by timmyomahony** [craftcms/redactor#171]
When using the "code" formatting option (which adds a `<pre>` tag) on a fresh Craft and Redactor install, the editor is collapsing new lines from my `<pre>` content.

Before:

<img width="1096" alt="Screenshot 2019-10-15 at 11 49 43" src="https://user-images.githubusercontent.com/615149/66826003-5934f180-ef43-11e9-82d9-581672a61a67.png">

After saving:

<img width="1085" alt="Screenshot 2019-10-15 at 11 49 50" src="https://user-images.githubusercontent.com/615149/66826012-5f2ad280-ef43-11e9-9e48-fc2b8f1b9889.png">

I would expect the editor to keep the line breaks in my code snippets but they seem to be being removed. Here is a plain HTML example:

<img width="612" alt="Screenshot 2019-10-15 at 12 17 31" src="https://user-images.githubusercontent.com/615149/66827041-ce092b00-ef45-11e9-8466-007f7bc4c1a6.png">

To try and fix it:

- I've tried adding the `"removeNewLines": false` option to my `.json` configuration.
- I've tried disabling HTML purify altogether
- I've unchecked all the advanced options for removing tags etc.

**Comment by timmyomahony** [comment]
OK, this seems like an issue going back to 2015:

https://craftcms.stackexchange.com/questions/6741/retain-empty-lines-in-a-richtext-field

**Comment by tinybeans** [comment]
@brandonkelly @timmyomahony In my opinion, this problem is caused by this commit (https://github.com/craftcms/redactor/commit/57331e6bbb028f3cbbe9560cdad97e0452ad8b00).


In my case (Craft CMS 3.3.11 and Redactor 2.4.0), after changing as follow, this line-break problem in `<pre>` was fixed.

Comment out `src/Field.php: line 362`

```
// $value = preg_replace('/[\r\n]/', '', $value);
```

Add a setting to my Redactor config

```
"removeNewLines": false,
```

**Comment by cherrykoda** [comment]
Im running into this as well, and its quite frustrating. There should be ability to enable or disable the newline stripping.

**Comment by andris-sevcenko** [comment]
Alright, so I just pushed a fix for this.

For full context:

* An initial solve for #161 came in the form of 57331e6bbb028f3cbbe9560cdad97e0452ad8b00. Not only this did not fully solve the problem, but it also, unbeknownst to us, also introduced the bug addressed by this issue.
* #161 was then fully fixed by craftcms/cms@602679ea534e23838ea03d3bb82552471ebc1d24
* That, also unbeknownst to us, solved the problem more effectively, rendering the previous commit redundant.

So, I basically reverted the original commit and am now taking full credit 😜

**Comment by andris-sevcenko** [comment]
This will be a part of the upcoming Redactor 2.6 release.
